pkg/kubecost/allocation_test.go

```diff
@@ -2068,8 +2068,254 @@ func TestAllocationSetRange_Accumulate(t *testing.T) {
 	}
 }
 
-// TODO niko/etl
-// func TestAllocationSetRange_AccumulateBy(t *testing.T) {}
+func TestAllocationSetRange_AccumulateBy(t *testing.T) {
+	ago3d := time.Now().UTC().Truncate(day).Add(-3 * day)
+	ago2d := time.Now().UTC().Truncate(day).Add(-2 * day)
+	yesterday := time.Now().UTC().Truncate(day).Add(-day)
+	today := time.Now().UTC().Truncate(day)
+	tomorrow := time.Now().UTC().Truncate(day).Add(day)
+	dur := time.Hour * 24 * 2
+
+	// Accumulating any combination of nil and/or empty set should result in empty set
+	// test 1 nil set
+	result, err := NewAllocationSetRange(nil).AccumulateBy(dur)
+	for _, as := range result.allocations {
+		if err != nil {
+			t.Fatalf("unexpected error accumulating nil AllocationSetRange: %s", err)
+		}
```

Comment:
This error check should be pulled out of the loop

Reply:
I'd also recommend using `Errorf` over `Fatalf` if later test logic can function even if this specific check fails. `Fatalf` causes an immediate exit while `Errorf` will continue running the test code but return this error. Because you've chosen to run multiple tests in one test function, I'd rewrite most of the `Fatalf`s. Then, if someone breaks multiple things, they get all the errors at once instead of one at a time :grin: 

https://pkg.go.dev/testing#T.Fatalf

Reply:
cool, thanks for the tip!